sdks/python/apache_beam/runners/interactive/pipeline_instrument.py

```diff
@@ -24,13 +24,27 @@
 from __future__ import absolute_import
 
 import apache_beam as beam
+from apache_beam.io.external.gcp.pubsub import ReadFromPubSub as ExternalReadFromPubSub
+from apache_beam.io.external.kafka import ReadFromKafka
+from apache_beam.io.gcp.bigquery_tools import BigQueryReader
+from apache_beam.io.gcp.pubsub import ReadFromPubSub
 from apache_beam.pipeline import PipelineVisitor
 from apache_beam.runners.interactive import cache_manager as cache
 from apache_beam.runners.interactive import interactive_environment as ie
 
 READ_CACHE = "_ReadCache_"
 WRITE_CACHE = "_WriteCache_"
 
+# Use a tuple to define the list of unbounded sources. It is not always feasible
+# to correctly find all the unbounded sources in the SDF world. This is
+# because SDF allows the source to dynamically create sources at runtime.
+REPLACEABLE_UNBOUNDED_SOURCES = (
+    ExternalReadFromPubSub,
+    ReadFromKafka,
+    ReadFromPubSub,
+    BigQueryReader,
```

Comment:
I don't think BigQueryReader is a streaming PTransform - or a PTransform at all. Other than that, LGTM.

Reply:
kk, removed

Reply:
Hm .. after thinking about it again - is this intended to be for local execution? Or should it work with other runners? (e.g. Dataflow)

If it's only for local execution, then External transforms won't work either.